diff --git a/arch/arm/mach-s3c2440/mach-mini2440.c b/arch/arm/mach-s3c2440/mach-mini2440.c
index fc2dc0b..7499dd1 100644
--- a/arch/arm/mach-s3c2440/mach-mini2440.c
+++ b/arch/arm/mach-s3c2440/mach-mini2440.c
@@ -228,24 +228,29 @@ static struct s3c24xx_mci_pdata mini2440_mmc_cfg __initdata = {
 static struct mtd_partition mini2440_default_nand_part[] __initdata = {
 	[0] = {
 		.name	= "u-boot",
-		.size	= SZ_256K,
+		.size	= 0x00060000,
 		.offset	= 0,
 	},
 	[1] = {
-		.name	= "u-boot-env",
-		.size	= SZ_128K,
-		.offset	= SZ_256K,
+		.name	= "u-boot_env",
+		.size	= 0x00020000,
+		.offset	= 0x00060000,
 	},
 	[2] = {
 		.name	= "kernel",
 		/* 5 megabytes, for a kernel with no modules
 		 * or a uImage with a ramdisk attached */
 		.size	= 0x00500000,
-		.offset	= SZ_256K + SZ_128K,
+		.offset	= 0x00080000,
 	},
 	[3] = {
-		.name	= "root",
-		.offset	= SZ_256K + SZ_128K + 0x00500000,
+		.name	= "splash",
+		.offset	= 0x00580000,
+		.size	= 0x00040000,
+	},
+	[4] = {
+		.name	= "rootfs",
+		.offset	= 0x005c0000,
 		.size	= MTDPART_SIZ_FULL,
 	},
 };
